File: yntra-ios/YntraIOS/Views/SettingsView.swift
import SwiftUI
import yntra_core

struct SettingsView: View {
    @ObservedObject var viewModel: SettingsViewModel
    var onLogout: () -> Void

    @State private var name: String = "Operator 1"
    @State private var phone: String = "+46 70 123 45 67"
    @State private var preferredLanguage: String = "sv"

    private let darkBackground = Color(red: 0.04, green: 0.06, blue: 0.1)
    private let cardBackground = Color(red: 0.12, green: 0.16, blue: 0.23)

    var body: some View {
        ZStack {
            darkBackground.ignoresSafeArea()
            
            VStack(spacing: 20) {
                // Header
                HStack {
                    Text("User Settings")
                        .font(.system(size: 28, weight: .black, design: .rounded))
                        .foregroundColor(.white)
                    Spacer()
                }
                .padding(.horizontal)
                .padding(.top, 10)

                if let errorMessage = viewModel.errorMessage {
                    Text("Error: \(errorMessage)")
                        .foregroundColor(.red)
                        .font(.caption)
                        .padding(.horizontal)
                }

                // Profile form card
                VStack(alignment: .leading, spacing: 16) {
                    Text("Edit Profile Info")
                        .font(.system(size: 16, weight: .bold, design: .rounded))
                        .foregroundColor(.white)
                    
                    VStack(alignment: .leading, spacing: 6) {
                        Text("Display Name")
                            .font(.caption)
                            .foregroundColor(.white.opacity(0.5))
                        TextField("", text: $name)
                            .padding()
                            .background(darkBackground)
                            .cornerRadius(12)
                            .foregroundColor(.white)
                    }

                    VStack(alignment: .leading, spacing: 6) {
                        Text("Phone Number")
                            .font(.caption)
                            .foregroundColor(.white.opacity(0.5))
                        TextField("", text: $phone)
                            .padding()
                            .background(darkBackground)
                            .cornerRadius(12)
                            .foregroundColor(.white)
                    }

                    // Select Language
                    VStack(alignment: .leading, spacing: 6) {
                        Text("Preferred Language")
                            .font(.caption)
                            .foregroundColor(.white.opacity(0.5))
                        
                        HStack(spacing: 6) {
                            ForEach([("sv", "SV"), ("da", "DA"), ("no", "NO"), ("fi", "FI"), ("en", "EN")], id: \.0) { code, label in
                                Button(action: { preferredLanguage = code }) {
                                    Text(label)
                                        .font(.system(size: 11, weight: .bold, design: .rounded))
                                        .foregroundColor(.white)
                                        .frame(maxWidth: .infinity)
                                        .padding(.vertical, 8)
                                        .background(preferredLanguage == code ? Color.purple : Color.white.opacity(0.1))
                                        .cornerRadius(8)
                                }
                            }
                        }
                    }

                    Button(action: {
                        viewModel.updateProfile(fullName: name, phone: phone, languagePreference: preferredLanguage)
                    }) {
                        Text("Save Profile Settings")
                            .bold()
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding()
                            .background(Color.blue)
                            .cornerRadius(12)
                    }
                }
                .padding(20)
                .background(cardBackground)
                .cornerRadius(20)
                .overlay(
                    RoundedRectangle(cornerRadius: 20)
                        .stroke(Color.white.opacity(0.05), lineWidth: 1)
                )
                .padding(.horizontal)

                // Workspace info panel
                VStack(alignment: .leading, spacing: 8) {
                    Text("Active Workspace Configuration")
                        .font(.system(size: 15, weight: .bold, design: .rounded))
                        .foregroundColor(.white)
                    
                    if let ws = viewModel.workspace {
                        Text("Name: \(ws.name)")
                            .font(.system(size: 13, weight: .semibold, design: .rounded))
                            .foregroundColor(.white.opacity(0.8))
                        Text("Active Modules: \(ws.modulesActive)")
                            .font(.system(size: 11, weight: .medium, design: .rounded))
                            .foregroundColor(.white.opacity(0.5))
                    } else {
                        Text("Loading config...")
                            .font(.caption)
                            .foregroundColor(.white.opacity(0.5))
                    }
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(20)
                .background(cardBackground)
                .cornerRadius(20)
                .overlay(
                    RoundedRectangle(cornerRadius: 20)
                        .stroke(Color.white.opacity(0.05), lineWidth: 1)
                )
                .padding(.horizontal)

                Spacer()

                // Logout button
                Button(action: onLogout) {
                    HStack(spacing: 8) {
                        Image(systemName: "power")
                        Text("Log Out Session")
                            .bold()
                    }
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .padding()
                    .background(Color.red)
                    .cornerRadius(16)
                }
                .padding(.horizontal)
                .padding(.bottom, 10)
            }
        }
    }
}

File: yntra-android/app/src/main/java/com/yntra/app/views/SettingsView.kt
package com.yntra.app.views

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ExitToApp
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.yntra.app.viewmodels.SettingsViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SettingsView(viewModel: SettingsViewModel, onLogout: () -> Unit) {
    val workspace by viewModel.workspace.collectAsState()
    val errorMessage by viewModel.errorMessage.collectAsState()

    var name by remember { mutableStateOf("Operator 1") }
    var phone by remember { mutableStateOf("+46 70 123 45 67") }
    var language by remember { mutableStateOf("sv") }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFF0B0F19))
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        // Header
        Text(
            text = "User Settings",
            color = Color.White,
            fontSize = 26.sp,
            fontWeight = FontWeight.Black
        )

        errorMessage?.let {
            Card(
                colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.errorContainer),
                shape = RoundedCornerShape(12.dp)
            ) {
                Text(
                    text = it,
                    color = MaterialTheme.colorScheme.onErrorContainer,
                    modifier = Modifier.padding(12.dp),
                    fontSize = 14.sp
                )
            }
        }

        // Profile details form card
        Card(
            colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
            shape = RoundedCornerShape(20.dp),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(
                modifier = Modifier.padding(20.dp),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Text(
                    text = "Edit Profile Info",
                    color = Color.White,
                    fontWeight = FontWeight.Bold,
                    fontSize = 16.sp
                )

                OutlinedTextField(
                    value = name,
                    onValueChange = { name = it },
                    label = { Text("Display Name") },
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(
                        focusedTextColor = Color.White,
                        unfocusedTextColor = Color.White
                    )
                )

                OutlinedTextField(
                    value = phone,
                    onValueChange = { phone = it },
                    label = { Text("Phone Number") },
                    modifier = Modifier.fillMaxWidth(),
                    colors = OutlinedTextFieldDefaults.colors(
                        focusedTextColor = Color.White,
                        unfocusedTextColor = Color.White
                    )
                )

                // Select Language
                Text(text = "Preferred Language", color = Color(0xFF94A3B8), fontSize = 12.sp, fontWeight = FontWeight.Bold)
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    listOf("sv" to "SV", "da" to "DA", "no" to "NO", "fi" to "FI", "en" to "EN").forEach { (code, label) ->
                        val isSelected = language == code
                        Box(
                            modifier = Modifier
                                .weight(1f)
                                .background(
                                    if (isSelected) Color(0xFF8B5CF6) else Color(0xFF334155),
                                    RoundedCornerShape(8.dp)
                                )
                                .clickable { language = code }
                                .padding(vertical = 8.dp),
                            contentAlignment = Alignment.Center
                        ) {
                            Text(text = label, color = Color.White, fontWeight = FontWeight.Bold, fontSize = 11.sp)
                        }
                    }
                }

                Button(
                    onClick = { viewModel.updateProfile(name, phone, language) },
                    modifier = Modifier.fillMaxWidth().padding(top = 8.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF4F46E5)),
                    shape = RoundedCornerShape(12.dp)
                ) {
                    Text("Save Profile Settings", fontWeight = FontWeight.Bold)
                }
            }
        }

        // Active Workspace details card
        Card(
            colors = CardDefaults.cardColors(containerColor = Color(0xFF1E293B)),
            shape = RoundedCornerShape(20.dp),
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(
                modifier = Modifier.padding(20.dp),
                verticalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                Text(
                    text = "Active Workspace Configuration",
                    color = Color.White,
                    fontWeight = FontWeight.Bold,
                    fontSize = 16.sp
                )
                
                workspace?.let { ws ->
                    Text(text = "Name: ${ws.name}", color = Color.White, fontSize = 14.sp)
                    Text(text = "Active Modules: ${ws.modulesActive}", color = Color(0xFF94A3B8), fontSize = 12.sp)
                } ?: Text(text = "Loading workspace config...", color = Color(0xFF94A3B8), fontSize = 13.sp)
            }
        }

        Spacer(modifier = Modifier.weight(1f))

        // Logout button
        Button(
            onClick = onLogout,
            colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFEF4444)),
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(16.dp),
            contentPadding = PaddingValues(16.dp)
        ) {
            Row(
                horizontalArrangement = Arrangement.spacedBy(8.dp),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Icon(imageVector = Icons.Default.ExitToApp, contentDescription = "Logout")
                Text("Log Out Session", fontSize = 16.sp, fontWeight = FontWeight.Bold)
            }
        }
    }
}

// Add simple clickable helper
private fun Modifier.clickable(onClick: () -> Unit): Modifier = this.clickable(
    onClick = onClick
)
